providers/adfs.go

```diff
@@ -0,0 +1,132 @@
+package providers
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/requests"
+)
+
+// ADFSProvider represents an ADFS based Identity Provider
+type ADFSProvider struct {
+	*OIDCProvider
+}
+
+var _ Provider = (*OIDCProvider)(nil)
+
+const (
+	adfsProviderName = "Adfs"
+	adfsDefaultScope = "openid"
```

Comment:
ADFS docs show it supports the `email` and `profile` scopes -- I think we should include those to align this with other OIDC providers:

```suggestion
	adfsDefaultScope = "openid email profile"
```